### PR TITLE
via_stack: accept gdsfactory's generic argument shape as primary

### LINKED ISSUE #217: via_stack argument shape diverges from gdsfactory generic via_stack
## Problem

`ihp.cells.via_stack` (ihp/cells/via_stacks.py) and gdsfactory's generic
`gdsfactory.components.vias.via_stack` (reused by most other PDKs) take
structurally different arguments for the same job:

- gdsfactory: `via_stack(size=(11,11), layers=("M1","M2","MTOP"), layer_offsets=None, vias=("via1","via2",None), layer_to_port_orientations=None, correct_size=False, slot_horizontal=False, slot_vertical=False, port_orientations=(180,90,0,-90))`
- IHP: `via_stack(bottom_layer="Metal1", top_layer="Metal2", size=(10,10), vn_columns=2, vn_rows=2, vt1_columns=1, vt1_rows=1, vt2_columns=1, vt2_rows=1, layer_activ=..., ...20 layer_* kwargs)`

Anyone porting code between IHP and other gdsfactory PDKs has to rewrite
every `via_stack(...)` call by hand.

## Proposal

Rework IHP's `via_stack` (ihp/cells/via_stacks.py) to accept gdsfactory's
argument names/order as primary, keeping IHP's bottom_layer/top_layer/
vn_*/vt*_* convenience as keyword-only extras after them.

## Mutually exclusive alternative

A companion issue/PR on gdsfactory/gdsfactory proposes the opposite
direction: teach the *generic* via_stack to accept bottom_layer/top_layer
IHP-style, with a deprecation fallback for the old positional call shape.
Only one of the two directions should ultimately land — see
https://github.com/gdsfactory/gdsfactory/issues/4641

### PR DESCRIPTION
**Pull request by thanojo**
Closes #217

Reworks `ihp.cells.via_stack` to accept size/layers/vias/... (the shape
gdsfactory's generic via_stack uses) as its primary arguments, keeping
bottom_layer/top_layer/vn_*/vt*_*/layer_* as keyword-only IHP extras.
Generic features IHP structurally can't support (vias, layer_offsets,
correct_size, slotting) raise NotImplementedError instead of silently
doing the wrong thing.

**Note on settings ref regeneration:** Adding new parameters to the
signature changes the kfactory cell-name hash and settings dump. The GDS
geometry is byte-identical; only the settings YAML was regenerated (per
CONTRIBUTING.md fallback provision). The new default cell name is
`via_stack_S10_10_LNone_LONone_VNone_LTPONone_CSFalse_SH_bcc180c1`.
Existing GDS files referencing the old default cell name string will need
updating.

**Mutually exclusive with gdsfactory/gdsfactory#4641 /
its companion PR** -- that direction teaches the *generic* via_stack to
accept IHP's shape instead. Only one of the two should land; this PR
stays in draft until that's decided.

## Summary by Sourcery

Rework the IHP via_stack cell API to use gdsfactory’s generic via_stack argument shape as the primary interface while keeping IHP-specific options as keyword-only extras.

New Features:
- Expose gdsfactory-style via_stack parameters (size, layers, vias, etc.) on ihp.cells.via_stack as primary arguments.
- Add a dedicated API test suite for via_stack covering default behavior, layers mapping, and unsupported generic kwargs.

Enhancements:
- Enforce explicit errors for generic via_stack features that IHP cannot support (vias, layer_offsets, correct_size, slotting, custom port orientations).
- Extend via_stack documentation to clarify supported/unsupported arguments and how generic layers map to IHP bottom/top layers.
- Regenerate via_stack-related settings snapshots and port data to match the revised API and default cell naming.

Documentation:
- Record the API change in the changelog describing the new primary argument shape for via_stack.

Tests:
- Add tests pinning the zero-argument via_stack settings and port scheme.
- Add tests verifying that the layers argument correctly maps to bottom_layer/top_layer.
- Add parameterized tests ensuring unsupported generic via_stack kwargs raise NotImplementedError.
- Add a test ensuring that mixing layers with explicit bottom_layer/top_layer raises ValueError.

### CODE REVIEW

**Comment by thanojo** [comment]
Mutually exclusive alternative direction: gdsfactory/gdsfactory#4642.

File: tests/test_via_stack_api.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        ("correct_size", True),
+        ("slot_horizontal", True),
+        ("slot_vertical", True),
+    ],
+)
+def test_unsupported_generic_kwargs_raise_not_implemented(kwarg, value) -> None:
+    """Generic features IHP's via_stack cannot honor must fail loudly, not silently."""
+    PDK.activate()
+    with pytest.raises(NotImplementedError):
+        via_stack(**{kwarg: value})
```
[the remark this answers is not in the document]
Comment (thanojo) [reply tests/test_via_stack_api.py:58]: adressed in [d8b8519](https://github.com/gdsfactory/IHP/pull/218/commits/d8b85192af3ca7c8e6aa65aadf214e60bb55cd03)

File: tests/test_via_stack_api.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    "kwarg,value",
+    [
+        ("vias", ("via1", "via2", None)),
+        ("layer_offsets", (0.0, 0.0)),
+        ("layer_to_port_orientations", {}),
+        ("correct_size", True),
+        ("slot_horizontal", True),
+        ("slot_vertical", True),
+    ],
+)
```
[the remark this answers is not in the document]
Comment (thanojo) [reply tests/test_via_stack_api.py:53]: adressed in [d8b8519](https://github.com/gdsfactory/IHP/pull/218/commits/d8b85192af3ca7c8e6aa65aadf214e60bb55cd03)
